src/Codeception/Module/WebDriver.php

```diff
@@ -976,17 +978,39 @@ public function dontSeeLink($text, $url = null)
             }
             return;
         }
+
+        $expectedUrl = $this->getAbsoluteUrlFor($url);
         $nodes = array_filter(
             $nodes,
-            function (WebDriverElement $e) use ($url) {
-                return trim($e->getAttribute('href')) == trim($url);
+            function (WebDriverElement $e) use ($expectedUrl) {
+                $elementHref = $this->getAbsoluteUrlFor($e->getAttribute('href'));
+                return $elementHref === $expectedUrl;
             }
         );
         if (!empty($nodes)) {
             $this->fail("Link containing text '$text' and URL '$url' was found in page " . $this->_getCurrentUri());
         }
     }
 
+
+    /**
+     * Returns an absolute URL for the passed URI with the current URL
+     * as the base path.
+     *
+     * @param string $uri the absolute or relative URI
+     * @return string the absolute URL
+     */
+    private function getAbsoluteUrlFor($uri)
```

Comment:
I will restructure this code because I don't like that `_getCurrentUri` makes HTTP​ request to Selenium on every call to `getAbsoluteUrlFor`.

I will introduce a private method `filterNodesByHref`  instead to reduce code duplication in `seeLink` and `dontSeeLink` methods.